test/extensions/filters/http/cache/cache_headers_utils_test.cc

```diff
@@ -0,0 +1,263 @@
+#include <chrono>
+#include <string>
+#include <vector>
+
+#include "envoy/common/time.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace {
+
+#define DURATION(s) std::chrono::seconds(s)
+#define UNSET_DURATION absl::optional<SystemTime::duration>()
+#define MAX_DURATION SystemTime::duration::max()
+
+struct RequestCacheControlTestCase {
+  absl::string_view cache_control_header;
+  RequestCacheControl request_cache_control;
+};
+
+struct ResponseCacheControlTestCase {
+  absl::string_view cache_control_header;
+  ResponseCacheControl response_cache_control;
+};
+
+class RequestCacheControlTest : public testing::TestWithParam<RequestCacheControlTestCase> {
+public:
+  // C++11 static members must be initialized out of line
+  const static RequestCacheControlTestCase test_cases[];
```

Comment:
Non-PoD static/global variables are forbidden. The Envoy style guide encourages following the advice at https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use. You can also look at other uses of INSTANTIATE_TEST_SUITE_P and copy their approaches.

Reply:
I will replace them with static functions returning the test cases.